### PR TITLE
lib: nrf_modem_lib: make irq priority configurable

### PR DESCRIPTION
**Pull request by SvenGrandcentrix**
As of now the IRQ priority is hard coded. The chosen IRQ level is illegal in case of enabled ZLI support. Adding this value to the KConfig seems to be the natural solution for the problem.

Signed-off-by: Sven Hädrich <sven.haedrich@grandcentrix.net>

### CODE REVIEW

File: lib/nrf_modem_lib/Kconfig.modemlib [hunk 1]
```diff
@@ -18,6 +18,10 @@ config NRF_MODEM_LIB_SYS_INIT
 	  Please note that initialization is synchronous and can take up to one
 	  minute in case the modem firmware is updated.
 
+config NRF_MODEM_TRACE_IRQ_PRIORITY
+	int "IRQ priority for trace handler"
+	default 6
```
Comment (rlubos) [inline lib/nrf_modem_lib/Kconfig.modemlib:23]: Would it make sense to set a different default in case ZLI is enabled, as 6 is not a valid one as you say?

**Review by rlubos — COMMENTED** [review]
Please add `lib: nrf_modem: ` prefix to the commit title and some description in the body.

File: lib/nrf_modem_lib/Kconfig.modemlib (hunk 1 above)
Comment (SvenGrandcentrix) [reply lib/nrf_modem_lib/Kconfig.modemlib:23]: An irq priority of 4 is compatible when ZLI is enabled. I suggest to make this the default value, than no intervention is necessary when ZLI is enabled. I feel uncomfortable to "automatically" change the priority based on CONFIG_ZERO_LATENCY_IRQS as this adds more "magic" than necessary.

**Comment by rlubos** [comment]
The commit message body has also a line length limit (72 char/line).

**Comment by SvenGrandcentrix** [comment]
fixed line breaks of commit messages.

**Comment by DatGizmo** [comment]
With the new Integration Test Specification process we don't need the CI-* github labels.
I will remove them.

To get the test spec feature working correctly on this PR, you need to rebase it to the latest main to get the [.github/test-spec.yml](https://github.com/nrfconnect/sdk-nrf/blob/main/.github/test-spec.yml) file

Then run CI here to update the test spec.

[Information about the test spec feature](https://projecttools.nordicsemi.no/confluence/display/NCS/Test+Spec)
[CI-Help for issues](https://ci-help.nordicsemi.no/)

**Pull request closed without merging**